--- Lib/xml/__init__.py
+++ Lib/xml/__init__.py
@@ -22,20 +22,22 @@
 _MINIMUM_XMLPLUS_VERSION = (0, 8, 4)
 
 
-try:
+def use_pyxml():
     import _xmlplus
-except ImportError:
-    pass
-else:
-    try:
-        v = _xmlplus.version_info
-    except AttributeError:
-        # _xmlplus is too old; ignore it
-        pass
+    v = _xmlplus.version_info
+    if v >= _MINIMUM_XMLPLUS_VERSION:
+        import sys
+        _xmlplus.__path__.extend(__path__)
+        sys.modules[__name__] = _xmlplus
+        modules = []
+        for module in sys.modules:
+            if module.startswith(("xml.__init__", "xml.dom", "xml.parsers", "xml.sax")) and sys.modules[module] is not None:
+                modules.append(module)
+        for module in modules:
+            del sys.modules[module]
+        for module in modules:
+            pyxml_module = module.replace("xml", "_xmlplus", 1)
+            __import__(pyxml_module)
+            sys.modules[module] = sys.modules[pyxml_module]
     else:
-        if v >= _MINIMUM_XMLPLUS_VERSION:
-            import sys
-            _xmlplus.__path__.extend(__path__)
-            sys.modules[__name__] = _xmlplus
-        else:
-            del v
+        raise ImportError("PyXML too old: %s" % ".".join(str(x) for x in v))
