--- configure.py
+++ configure.py
@@ -1377,8 +1377,10 @@
         sipconfig.inform("DBus v1 does not seem to be installed.")
         return
 
-    # Using str() means it will work with both Python v2 and v3.
-    for f in str(iflags).split():
+    if sys.hexversion >= 0x03000000:
+        iflags = iflags.decode()
+
+    for f in iflags.split():
         if f.startswith("-I"):
             dbusincdirs.append(f[2:])
         elif f.startswith("-L"):
--- dbus/dbus.cpp
+++ dbus/dbus.cpp
@@ -1,7 +1,7 @@
 // This is the SIP interface definition for the Qt support for the standard
 // Python DBus bindings.
 //
-// Copyright (c) 2007 Phil Thompson
+// Copyright (c) 2012 Phil Thompson
 //
 // Licensed under the Academic Free License version 2.1
 //
@@ -392,6 +392,24 @@
 
 
 // The module entry point.
+#if PY_MAJOR_VERSION >= 3
+PyMODINIT_FUNC PyInit_qt()
+{
+    static PyModuleDef module_def = {
+        PyModuleDef_HEAD_INIT,
+        "qt",
+        NULL,
+        -1,
+        module_functions,
+    };
+
+    // Import the generic part of the Python DBus bindings.
+    if (import_dbus_bindings("dbus.mainloop.qt") < 0)
+        return 0;
+
+    return PyModule_Create(&module_def);
+}
+#else
 PyMODINIT_FUNC initqt()
 {
     // Import the generic part of the Python DBus bindings.
@@ -400,3 +418,4 @@
 
     Py_InitModule("qt", module_functions);
 }
+#endif
