http://bazaar.launchpad.net/~testtools-committers/testtools/trunk/revision/244

--- testtools/tests/test_deferredruntest.py
+++ testtools/tests/test_deferredruntest.py
@@ -13,7 +13,6 @@
 from testtools.content import (
     text_content,
     )
-from testtools.deferredruntest import run_with_log_observers
 from testtools.helpers import try_import
 from testtools.matchers import (
     Equals,
@@ -747,9 +746,10 @@
             lambda x: self.fail("Should not have succeeded"), check_result)
 
 
-class TestRunWithLogObservers(TestCase):
+class TestRunWithLogObservers(NeedsTwistedTestCase):
 
     def test_restores_observers(self):
+        from testtools.deferredruntest import run_with_log_observers
         from twisted.python import log
         # Make sure there's at least one observer.  This reproduces bug
         # #926189.
--- testtools/tests/test_fixturesupport.py
+++ testtools/tests/test_fixturesupport.py
@@ -7,7 +7,7 @@
     content,
     content_type,
     )
-from testtools.compat import _b
+from testtools.compat import _b, _u
 from testtools.helpers import try_import
 from testtools.testresult.doubles import (
     ExtendedTestResult,
@@ -89,8 +89,8 @@
         self.assertEqual('addSuccess', result._events[-2][0])
         details = result._events[-2][2]
         self.assertEqual(['aaa', 'bbb'], sorted(details))
-        self.assertEqual(u'foo', details['aaa'].as_text())
-        self.assertEqual(u'bar', details['bbb'].as_text())
+        self.assertEqual(_u('foo'), details['aaa'].as_text())
+        self.assertEqual(_u('bar'), details['bbb'].as_text())
 
     def test_useFixture_details_captured_from_setUp(self):
         # Details added during fixture set-up are gathered even if setUp()
