https://github.com/waylan/Python-Markdown/issues/203

--- markdown/blockprocessors.py
+++ markdown/blockprocessors.py
@@ -1,17 +1,19 @@
-from __future__ import unicode_literals
-# CORE MARKDOWN BLOCKPARSER
-# ===========================================================================
-#
-# This parser handles basic parsing of Markdown blocks.  It doesn't concern itself
-# with inline elements such as **bold** or *italics*, but rather just catches 
-# blocks, lists, quotes, etc.
-#
-# The BlockParser is made up of a bunch of BlockProssors, each handling a 
-# different type of block. Extensions may add/replace/remove BlockProcessors
-# as they need to alter how markdown blocks are parsed.
+"""
+CORE MARKDOWN BLOCKPARSER
+===========================================================================
+
+This parser handles basic parsing of Markdown blocks.  It doesn't concern itself
+with inline elements such as **bold** or *italics*, but rather just catches
+blocks, lists, quotes, etc.
+
+The BlockParser is made up of a bunch of BlockProssors, each handling a
+different type of block. Extensions may add/replace/remove BlockProcessors
+as they need to alter how markdown blocks are parsed.
+"""
 
 from __future__ import absolute_import
 from __future__ import division
+from __future__ import unicode_literals
 import logging
 import re
 from . import util
--- markdown/extensions/abbr.py
+++ markdown/extensions/abbr.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 '''
 Abbreviation Extension for Python-Markdown
 ==========================================
@@ -25,6 +24,7 @@
 '''
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..preprocessors import Preprocessor
 from ..inlinepatterns import Pattern
--- markdown/extensions/admonition.py
+++ markdown/extensions/admonition.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Admonition extension for Python-Markdown
 ========================================
@@ -42,6 +41,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..blockprocessors import BlockProcessor
 from ..util import etree
--- markdown/extensions/attr_list.py
+++ markdown/extensions/attr_list.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Attribute List Extension for Python-Markdown
 ============================================
@@ -20,6 +19,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..treeprocessors import Treeprocessor
 from ..util import isBlockLevel
--- markdown/extensions/codehilite.py
+++ markdown/extensions/codehilite.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 CodeHilite Extension for Python-Markdown
 ========================================
@@ -20,6 +19,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..treeprocessors import Treeprocessor
 import warnings
--- markdown/extensions/def_list.py
+++ markdown/extensions/def_list.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Definition List Extension for Python-Markdown
 =============================================
@@ -20,6 +19,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..blockprocessors import BlockProcessor, ListIndentProcessor
 from ..util import etree
--- markdown/extensions/extra.py
+++ markdown/extensions/extra.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Python-Markdown Extra Extension
 ===============================
@@ -28,6 +27,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 
 extensions = ['smart_strong',
--- markdown/extensions/fenced_code.py
+++ markdown/extensions/fenced_code.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Fenced Code Extension for Python Markdown
 =========================================
@@ -75,6 +74,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..preprocessors import Preprocessor
 from .codehilite import CodeHilite, CodeHiliteExtension
--- markdown/extensions/footnotes.py
+++ markdown/extensions/footnotes.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 ========================= FOOTNOTES =================================
 
@@ -25,6 +24,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..preprocessors import Preprocessor
 from ..inlinepatterns import Pattern
--- markdown/extensions/headerid.py
+++ markdown/extensions/headerid.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 HeaderID Extension for Python-Markdown
 ======================================
@@ -76,6 +75,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..treeprocessors import Treeprocessor
 import re
--- markdown/extensions/__init__.py
+++ markdown/extensions/__init__.py
@@ -1,9 +1,10 @@
-from __future__ import unicode_literals
 """
 Extensions
 -----------------------------------------------------------------------------
 """
 
+from __future__ import unicode_literals
+
 class Extension(object):
     """ Base class for extensions to subclass. """
     def __init__(self, configs = {}):
--- markdown/extensions/meta.py
+++ markdown/extensions/meta.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Meta Data Extension for Python-Markdown
 =======================================
@@ -41,6 +40,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..preprocessors import Preprocessor
 import re
--- markdown/extensions/nl2br.py
+++ markdown/extensions/nl2br.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 NL2BR Extension
 ===============
@@ -22,6 +21,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..inlinepatterns import SubstituteTagPattern
 
--- markdown/extensions/sane_lists.py
+++ markdown/extensions/sane_lists.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Sane List Extension for Python-Markdown
 =======================================
@@ -20,6 +19,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..blockprocessors import OListProcessor, UListProcessor
 import re
--- markdown/extensions/smart_strong.py
+++ markdown/extensions/smart_strong.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 '''
 Smart_Strong Extension for Python-Markdown
 ==========================================
@@ -24,6 +23,7 @@
 '''
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..inlinepatterns import SimpleTagPattern
 
--- markdown/extensions/tables.py
+++ markdown/extensions/tables.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Tables Extension for Python-Markdown
 ====================================
@@ -16,6 +15,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..blockprocessors import BlockProcessor
 from ..util import etree
--- markdown/extensions/toc.py
+++ markdown/extensions/toc.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Table of Contents Extension for Python-Markdown
 * * *
@@ -11,6 +10,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..treeprocessors import Treeprocessor
 from ..util import etree
--- markdown/extensions/wikilinks.py
+++ markdown/extensions/wikilinks.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 '''
 WikiLinks Extension for Python-Markdown
 ======================================
@@ -78,6 +77,7 @@
 '''
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import Extension
 from ..inlinepatterns import Pattern
 from ..util import etree
--- markdown/__init__.py
+++ markdown/__init__.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 Python Markdown
 ===============
@@ -32,6 +31,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from .__version__ import version, version_info
 import re
 import codecs
--- markdown/inlinepatterns.py
+++ markdown/inlinepatterns.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 INLINE PATTERNS
 =============================================================================
@@ -43,6 +42,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import util
 from . import odict
 import re
--- markdown/postprocessors.py
+++ markdown/postprocessors.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 POST-PROCESSORS
 =============================================================================
@@ -10,6 +9,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import util
 from . import odict
 import re
--- markdown/preprocessors.py
+++ markdown/preprocessors.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 """
 PRE-PROCESSORS
 =============================================================================
@@ -8,6 +7,7 @@
 """
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import util
 from . import odict
 import re
--- markdown/serializers.py
+++ markdown/serializers.py
@@ -1,4 +1,3 @@
-from __future__ import unicode_literals
 # markdown/searializers.py
 #
 # Add x/html serialization to Elementree
@@ -39,6 +38,7 @@
 
 
 from __future__ import absolute_import
+from __future__ import unicode_literals
 from . import util
 ElementTree = util.etree.ElementTree
 QName = util.etree.QName
